### ISSUE TITLE
thread 'main' panicked at 'no entry found for key': lalrpop-0.19.7/src/lr1/lane_table/construct/merge.rs:202:20

### ISSUE THREAD

**Issue by aDotInTheVoid** [lalrpop/lalrpop#665]
With this grammar:

```rust
grammar();

NestedExpression: () = { 
    "N",
    "(" Expression ")"
}

IndexingExpression: () = {
	<Expression> "[" "]",
}

ObjectSelector: () = {
	"N",
	IndexingExpression,
}

VariadicExpression: () = {
	"vaarg" "(" <ObjectSelector> ")"
}

BuildinExpression: () = {
	NestedExpression,
	VariadicExpression,
}

Assignment: () = {
	ObjectSelector Expression,
}

pub Expression: () = { 
	Assignment,
	NestedExpression ,
	VariadicExpression ,
}
```

Running it gives:

```shell
RUST_BACKTRACE=1 lalrpop src/gr.lalrpop
processing file `src/gr.lalrpop`
thread 'main' panicked at 'no entry found for key', /home/nixon/.cargo/registry/src/github.com-1ecc6299db9ec823/lalrpop-0.19.7/src/lr1/lane_table/construct/merge.rs:202:20
stack backtrace:
   0: rust_begin_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
   2: core::panicking::panic_display
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:73:5
   3: core::panicking::panic_str
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:56:5
   4: core::option::expect_failed
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/option.rs:1852:5
   5: lalrpop::lr1::lane_table::construct::merge::ContextSets::union
   6: lalrpop::lr1::lane_table::construct::merge::Merge::walk
   7: lalrpop::lr1::lane_table::construct::merge::Merge::walk
   8: lalrpop::lr1::lane_table::construct::merge::Merge::start
   9: lalrpop::lr1::lane_table::build_lane_table_states
  10: lalrpop::lr1::build::build_lr1_states
  11: lalrpop::lr1::build_states
  12: lalrpop::build::process_file_into
  13: lalrpop::build::process_file
  14: lalrpop::api::Configuration::process_file
  15: lalrpop::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

While this grammar is nonsensical, is is a minification of a [much larger one](https://gist.github.com/aDotInTheVoid/498579cb4b485dcd0bfcf30f1984941d), that should (AFAIKT) compille and be unambiguous, which may be helpful as a test case

**Comment by osa1** [comment]
The short repro above no longer panics with the current master branch (b777a6a), but the linked large grammar still panics. Backtrace with the debug build:

```
thread 'main' panicked at 'no entry found for key', lalrpop/src/lr1/lane_table/construct/merge.rs:202:20
stack backtrace:
   0: rust_begin_unwind
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/panicking.rs:142:14
   2: core::panicking::panic_display
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/panicking.rs:72:5
   3: core::panicking::panic_str
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/panicking.rs:56:5
   4: core::option::expect_failed
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/option.rs:1874:5
   5: core::option::Option<T>::expect
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/option.rs:738:21
   6: <alloc::collections::btree::map::BTreeMap<K,V,A> as core::ops::index::Index<&Q>>::index
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/alloc/src/collections/btree/map.rs:2197:9
   7: lalrpop::lr1::lane_table::construct::merge::ContextSets::union
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/lane_table/construct/merge.rs:202:20
   8: lalrpop::lr1::lane_table::construct::merge::Merge::walk
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/lane_table/construct/merge.rs:90:16
   9: lalrpop::lr1::lane_table::construct::merge::Merge::start
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/lane_table/construct/merge.rs:52:9
  10: lalrpop::lr1::lane_table::construct::LaneTableConstruct::resolve_inconsistencies
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/lane_table/construct/mod.rs:201:19
  11: lalrpop::lr1::lane_table::construct::LaneTableConstruct::construct
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/lane_table/construct/mod.rs:75:19
  12: lalrpop::lr1::lane_table::build_lane_table_states
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/lane_table/mod.rs:17:5
  13: lalrpop::lr1::build::build_lr1_states
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/build/mod.rs:50:13
  14: lalrpop::lr1::build_states
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/lr1/mod.rs:32:9
  15: lalrpop::build::emit_recursive_ascent
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/build/mod.rs:386:25
  16: lalrpop::build::process_file_into
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/build/mod.rs:134:26
  17: lalrpop::build::process_file
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/build/mod.rs:68:5
  18: lalrpop::api::Configuration::process_file
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/api/mod.rs:244:9
  19: lalrpop::main1
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/main.rs:156:15
  20: lalrpop::main
             at /usr/local/google/home/omersa/rust/lalrpop/lalrpop/src/main.rs:90:5
  21: core::ops::function::FnOnce::call_once
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by segeljakt** [comment]
I got the same error with this code:
```
#[inline] SeparatedLList<S, X>: Vector<X> = {
    Epsilon => vector![],
    SeparatedNonemptyLList<S, X>
}

SeparatedNonemptyLList<S, X>: Vector<X> = {
    X => vector![<>],
    <mut l:SeparatedNonemptyLList<S, X>> S <x:X> => { l.push_back(x); l }
}

#[inline] SeqExplicit<X>: Vector<X> = <x:X> "," <mut l:SeparatedLList<",", X>> => { l.push_front(x); l };

Spanned<X>: Spanned<X> = <l:@L> <x:X> <r:@R> => (x, Info::new(id, l, r));
Pattern<P>: Pattern = Spanned<P> => (<>.0).with(<>.1);

P0: PatternKind = {
    P1
}

P1: PatternKind = {
    P2
}

P2: PatternKind = {
    Paren<Pattern<P1>> => PParen(<>), // Changing P1 to P0 makes it work
    Tuple<Pattern<P0>> => PTuple(<>),
}

Paren<X> = "(" <X> ")";
Tuple<X>  = Paren<SeqExplicit<X>>;
```

**Comment by dburgener** [comment]
This should be fixed by #898.  Thanks for the report!  If you're still seeing issues after the next release (or when building the lastest code from github) please reopen.
